website/client/components/shops/buyModal.vue

```diff
@@ -285,6 +285,11 @@
 
   import moment from 'moment';
 
+  const hideAmountSelectionForPurchaseTypes = [
```

Comment:
I think @TheHollidayInn implemented a partial fix for this here https://github.com/HabitRPG/habitica/pull/9519/commits/8664243ca9e35f22289187a848b375a03d016a53#diff-b66c19ca9da748e74190b22776c587b7L45, this fix seems broader but you'll want to remove his code when this is merged